api/gauge_agg.md

```diff
@@ -0,0 +1,76 @@
+# gauge_agg() <tag type="toolkit" content="Toolkit" /><tag type="experimental" content="Experimental" />
+Produces a `GaugeSummary` that can be used to accumulate gauge data for further
+calculations. 
+```sql
+gauge_agg (
+    ts TIMESTAMPTZ,
+    value DOUBLE PRECISION
+) RETURNS GaugeSummary
+```
+
+<highlight type="warning">
+Experimental features could have bugs. They might not be backwards compatible,
+and could be removed in future releases. Use these features at your own risk,
+and do not use any experimental features in production.
+</highlight>
+
+For more information about counter and gauge aggregation functions, see the
+[hyperfunctions documentation][hyperfunctions-counter-agg].
+
+## Required arguments
+
+|Name|Type|Description
+|-|-|-|
+|`ts`|`TIMESTAMPTZ`|The time at each point|
+|`value`|`DOUBLE PRECISION`|The value at that timestamp|
+
+Only `DOUBLE PRECISION` values are accepted for the `value` parameter. For gauge
+data stored as other numeric types, cast it to `DOUBLE PRECISION` when using the
+function.
+
+<highlight type="note">
+Both `ts` and `value` can be `NULL`, but the aggregate is not evaluated on
+`NULL` values. If the aggregate receives only a `NULL` value, it returns
+`NULL`. If it receives both non-`NULL` and `NULL` values, it ignores the `NULL`
```

Comment:
While this may be our goal, it doesn't seem to be true for any of our aggregates any longer due to https://github.com/timescale/timescaledb-toolkit/issues/362

I just added a test for state_agg and can confirm it is suffering the same bug.

Reply:
Answered my own question :) I tested with `counter_agg` and it worked if some values were `NULL` but not if all values were `NULL`